apache/superset
# feat: Add Area chart migration and tweaks the Timeseries chart migration

### SUMMARY
This PR adds the Area chart migration logic (NVD3 ➡️ ECharts) and tweaks the base class used by timeseries migrations. Users can execute this migration using the https://github.com/apache/superset/pull/25304 CLI command and disable the legacy version with the [VIZ_TYPE_DENYLIST](https://github.com/apache/superset/blob/ec6191023263ac5f8292dd37f037805c3196e029/superset/config.py#L829) configuration.

@sadpandajoe @jinghua-qa

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<img width="1440" alt="Screenshot 2023-11-10 at 14 28 21" src="https://github.com/apache/superset/assets/70410625/507172c9-6b2b-4df7-bd74-4629295cbd88">
<img width="1443" alt="Screenshot 2023-11-10 at 14 28 46" src="https://github.com/apache/superset/assets/70410625/c1100a93-142c-4dd7-bf2d-ec4b7b39f23c">

### TESTING INSTRUCTIONS
1 - Upgrade a Area chart using the CLI command
2 - Check the new chart
3 - Downgrade a Area chart using the CLI command
4 - Check the legacy chart

### ADDITIONAL INFORMATION
